www/tesseract.html

```diff
@@ -0,0 +1,72 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <title>Warp Wasm Test</title>
+  </head>
+  <script type="module">
+    import init, * as wasm from './wasm/warp/warp.js';
+    init().then((_exports) => {
+      console.log(wasm)
+
+      let tesseract = new wasm.Tesseract()
+      tesseract.load_from_storage()
+      console.log(`loaded data`)
+
+      const passphrase = new Uint8Array([1,2,3,4,5,6,7,8,9,0]);
+      //self.crypto.getRandomValues(passphrase);
+      console.log(`passphrase: ${passphrase}`);
+      tesseract.unlock(passphrase)
+      console.log(`unlocked`)
+
+      if (!tesseract.autosave_enabled()) {
+        tesseract.set_autosave()
+      }
+      console.log(`enabled saving`)
+
+      let key = `mykey`
+      if (tesseract.exist(key)) {
+        console.log(`${key} exists: ${tesseract.retrieve(key)}`)
+      } else {
+        console.log(`${key} does not exist yet`)
+      }
+
+      tesseract.set(key, `value123`)
+      console.log(`set ${key}: ${tesseract.retrieve(key)}`)
+      tesseract.set(key, `persisted123`)
+      console.log(`set ${key}: ${tesseract.retrieve(key)}`)
+
+      tesseract.save()
+      console.log(`saved: ${key}`)
+
+      let stream = tesseract.subscribe()
+      console.log(`subscribed to stream`)
+
+      async function stream_reader() {
+        while (true) {
+          console.log(wasm.TesseractEvent[await stream.next()])
+        }
+      }
+      stream_reader()
+      console.log(`started stream reader loop`)
+
+      async function tesseract_lock_loop() {
+        for (let i = 0; i < 10; i++) {
+          await new Promise(resolve => setTimeout(resolve, 50))
+          tesseract.lock()
+        }
+      }
+      tesseract_lock_loop()
+      console.log(`started tesseract lock loop`)
+
+    });
+  </script>
+  <body>
+    <h2>How to use this example:</h2>
+    <h3>1 - Make the sure wasm is compiled:</h3>
+    <pre>
+      wasm-pack build tools/wasm-example --target web --out-dir www/wasm/wasm-example
+    </pre>
+    <h3>2 - Check ouput in the console</h3>
+  </body>
+</html>
```

Comment:
Can be done in a separate PR but the `www` directory should probably be moved to the wasm example directory in `tools`

Reply:
done